test-framework/test-suites/unit/tests/pylib/stack/test_pylib_stack_password.py

```diff
@@ -0,0 +1,12 @@
+from stack.password import Password
+
+from unittest.mock import patch
+
+class TestPassword:
+	@patch("os.getrandom")
+	@patch("time.time")
+	def test_get_rand_blocking_io(self, mock_getrandom, mock_time):
```

Comment:
With multiple nested `patch` decorators, the mocked object is passed into the function in the same order they are applied. Since python decorators accept functions as their args, they are applied inside out. Basically, what Im trying to say is, the argument defined `mock_getrandom` is actually a mock of `time.time` and `mock_time` is actually the mock of `os.getrandom`. Which makes sense with your test since you would expect `os.getrandom` to get called 16 times.

Reply:
Good catch! That was driving me nuts. I'll fix it, remove the #pragma and see if my exception block actually gets ran.

Reply:
Cool, my test actually is covering my targeted code path. Thanks!